Objects/codeobject.c

```diff
@@ -1478,6 +1488,20 @@ deopt_code(PyCodeObject *code, _Py_CODEUNIT *instructions)
         }
         i += caches;
     }
+    if (code->co_executors == NULL) {
+        return;
+    }
+    for (int i = 0; i < len; i++) {
+        int opcode = instructions[i].op.code;
+        if (opcode == ENTER_EXECUTOR) {
+            _PyExecutorObject *exec = code->co_executors->executors[instructions[i].op.arg];
+            opcode = instructions[i].op.code = exec->vm_data.opcode;
+            instructions[i].op.arg = exec->vm_data.oparg;
+            assert(instructions[i+1].cache < (1 << OPTIMIZER_BITS_IN_COUNTER));
```

Comment:
We need to zero the caches here, since we missed them the first time around.

Maybe this could be done before (or as part of) the loop above?